beacon-chain/state/state-native/beacon_state_mainnet.go

```diff
@@ -60,3 +62,114 @@ type BeaconState struct {
+	InactivityScores                    []uint64                                `json:"inactivity_scores" yaml:"inactivityScores"`
+	CurrentSyncCommittee                *ethpb.SyncCommittee                    `json:"current_sync_committee" yaml:"currentSyncCommittee"`
+	NextSyncCommittee                   *ethpb.SyncCommittee                    `json:"next_sync_committee" yaml:"nextSyncCommittee"`
+	LatestExecutionPayloadHeader        *enginev1.ExecutionPayloadHeader        `json:"latest_execution_payload_header" yaml:"latestExecutionPayloadHeader"`
+	LatestExecutionPayloadHeaderCapella *enginev1.ExecutionPayloadHeaderCapella `json:"latest_execution_payload_header_capella" yaml:"latestExecutionPayloadHeaderCapella"`
+	NextWithdrawalIndex                 uint64                                  `json:"next_withdrawal_index" yaml:"nextWithdrawalIndex"`
+	NextWithdrawalValidatorIndex        primitives.ValidatorIndex               `json:"next_withdrawal_validator_index" yaml:"nextWithdrawalValidatorIndex"`
+}
+
+func (b *BeaconState) MarshalJSON() ([]byte, error) {
+	marshalable := &beaconStateMarshalable{
+		Version:                             b.version,
+		GenesisTime:                         b.genesisTime,
+		GenesisValidatorsRoot:               b.genesisValidatorsRoot,
+		Slot:                                b.slot,
+		Fork:                                b.fork,
+		LatestBlockHeader:                   b.latestBlockHeader,
+		BlockRoots:                          b.blockRoots,
+		StateRoots:                          b.stateRoots,
+		HistoricalRoots:                     b.historicalRoots,
+		HistoricalSummaries:                 b.historicalSummaries,
+		Eth1Data:                            b.eth1Data,
+		Eth1DataVotes:                       b.eth1DataVotes,
+		Eth1DepositIndex:                    b.eth1DepositIndex,
+		Validators:                          b.validators,
+		Balances:                            b.balances,
+		RandaoMixes:                         b.randaoMixes,
+		Slashings:                           b.slashings,
+		PreviousEpochAttestations:           b.previousEpochAttestations,
+		CurrentEpochAttestations:            b.currentEpochAttestations,
+		PreviousEpochParticipation:          b.previousEpochParticipation,
+		CurrentEpochParticipation:           b.currentEpochParticipation,
+		JustificationBits:                   b.justificationBits,
+		PreviousJustifiedCheckpoint:         b.previousJustifiedCheckpoint,
+		CurrentJustifiedCheckpoint:          b.currentJustifiedCheckpoint,
+		FinalizedCheckpoint:                 b.finalizedCheckpoint,
+		InactivityScores:                    b.inactivityScores,
+		CurrentSyncCommittee:                b.currentSyncCommittee,
+		NextSyncCommittee:                   b.nextSyncCommittee,
+		LatestExecutionPayloadHeader:        b.latestExecutionPayloadHeader,
+		LatestExecutionPayloadHeaderCapella: b.latestExecutionPayloadHeaderCapella,
```

Comment:
you will probably need to do this by fork or something see this is only for capella and will need to be updated for deneb. is there a way you can use the existing datastructure elsewhere?

Reply:
I think this object might live somewhere else as well too 🤔